src/ejabberd_service.erl

```diff
@@ -246,6 +246,9 @@ wait_for_handshake({xmlstreamend, _Name}, StateData) ->
 wait_for_handshake({xmlstreamerror, _}, StateData) ->
     send_text(StateData, <<(mongoose_xmpp_errors:xml_not_well_formed_bin())/binary, (?STREAM_TRAILER)/binary>>),
     {stop, normal, StateData};
+wait_for_handshake(replaced, StateData) ->
+    %% We don't expect to receive replaced before handshake
```

Comment:
What about line 224? We handle `replaced` before `handshake` there. :D

Reply:
224: just drop connection
250: close stream by sending stream:error

Reply:
I see now that I actually misunderstood the comment. It's fine.